pkg/csi/service/common/commonco/coagnostic.go

```diff
@@ -66,6 +66,8 @@ type COCommonInterface interface {
 	GetAllVolumes() []string
 	// GetAllK8sVolumes returns list of volumes in a bound state, in the K8s cluster
 	GetAllK8sVolumes() []string
+	// NodeNameMatchesInPodSpec returns true when the given nodeName and volumeID matches with a Pod Spec.
+	NodeNameMatchesInPodSpec(ctx context.Context, volumeHandle string, nodeNameInDetachRequest string) bool
```

Comment:
I think we should refactor this to `GetNodeForAttachedVolume` or something similar, keeping consistent with the design choice that coagnostic interface should be Kubernetes agnostic. From k8sOrchestrator we can just return the node for the pod, and check in WCP controller if this node is same as that in request.

Thinking on similar lines, I think we should also refactor the method `GetVolumeAttachment` introduced earlier to `GetAttachedVMFortheVolume` or something like that. This will keep the interface CO agnostic. The idea is to get the VM and see if the volume is still attached to that VM.

Let me know if this makes sense. Also I came across `GetNodesForVolumes` method introduced in the interface for `ListVolumes` feature. See if we can make use of it, instead of iterating through all the pods for the pvc.

Reply:
Renamed the method to - `GetNodeForAttachedVolume`. 

`GetNodesForVolumes` does not seem to be consistent way of checking nodeID as the map is updated & removed based on Informers event. I am worried that csi container restart may invalidate the maps.

Reply:
Makes sense.